### PR TITLE
Gambit user signup inegration

### PR DESCRIPTION
**Pull request by sergiitk**
#### What's this PR do?
- Implements [posting user signup](https://github.com/DoSomething/gambit/blob/develop/documentation/endpoints/signups.md) to Gambit
- Implements [Gambit authorization](https://github.com/DoSomething/gambit/wiki/Chatbot)
#### Where should the reviewer start?
- @weerd or @DFurnes Please review integration with Gateway
- @aaronschachter Please review integration with Gambit
- @deadlybutter CC
#### What are the relevant tickets?

Ref DoSomething/mbc-registration-email#71.

### CODE REVIEW

File: src/Gambit.php [hunk 1]
```diff
@@ -8,6 +8,13 @@
 
 class Gambit extends RestApiClient
 {
+    use AuthorizesWithGambit;
+
+    /**
+     * Unknown signup source.
+     */
+    const SIGNUP_SOURCE_FALLBACK = 'unknown';
```
Comment (aaronschachter) [inline src/Gambit.php:16]: 🌀

**Review by aaronschachter — APPROVED** [review]

File: src/Gambit.php [hunk 2]
```diff
@@ -47,12 +66,41 @@ public function getAllCampaigns()
...
 
         if (is_null($response)) {
             return null;
         }
 
         return new GambitCampaign($response['data']);
     }
+
+    /**
+     * Send a Post request Gambit signup endpoing.
```
Comment (DFurnes) [inline src/Gambit.php]: typo `endpoing` → `endpoint`

**Review by DFurnes — APPROVED** [review]

**Review by weerd — APPROVED** [review]
👍

File: src/Gambit.php (hunk 2 above)
Comment (sergiitk) [reply src/Gambit.php]: Thanks!

**Pull request merged**